artifact/tar_writer_stream.go

```diff
@@ -43,6 +43,9 @@ func NewTarWriterStream(w *tar.Writer) *StreamArchiver {
 }
 
 func (str *StreamArchiver) Write(data []byte, archivePath string) error {
+	if str.Writer == nil {
+		return errors.New("arch: Can not write to empty tar-writer")
```

Comment:
I think it is better to return an error than to panic on an empty tar writer.

Reply:
:+1: